File: refactoring-practice-1/src/main/java/dojo/replace_data_value_with_object/OrderUtils.java
package dojo.replace_data_value_with_object;

public class OrderUtils {

	public static int getNumberOfOrdersFor(String customerName, Order ... orders) {
		int result = 0;
		for (Order order : orders) {
			if (order.getCustomer().equals(customerName)) {
				result++;
			}
		}
		return result;
	}
	
}

File: refactoring-practice-1/src/main/typescript/replace_data_value_with_object/order-utils.ts
import {Order} from "./order.js";

export class OrderUtils {

    public static getNumberOfOrdersFor(customerName: string, orders: Order[]) {
        let result = 0;
        for (let order of orders) {
            if (order.getCustomerName() === customerName) {
                result++;
            }
        }
        return result;
    }

}
